### PR TITLE
exchanged code containing goto with a do while loop

### PR DESCRIPTION
**Pull request by renepickhardt**
did not find anything in the coding style which would require the use of goto therefor I thought it can be exchanged with a `do {...} while()` statement

### CODE REVIEW

File: lightningd/watch.c [hunk 1]
```diff
@@ -270,19 +270,17 @@ void watch_topology_changed(struct chain_topology *topo)
...
+		for (w = txwatch_hash_first(&topo->txwatches, &i);
+	    	w;
+	    	w = txwatch_hash_next(&topo->txwatches, &i)) {
+				u32 depth;
+
+				depth = get_tx_depth(topo, &w->txid);
+				if (depth)
+					needs_rerun |= txw_fire(w, &w->txid, depth);
+		}
+	}while (needs_rerun);
```
Comment (cdecker) [inline lightningd/watch.c]: Nit: whitespace befor the `while` is missing, and after the `do`.

File: lightningd/watch.c [hunk 2]
```diff
@@ -270,19 +270,17 @@ void watch_topology_changed(struct chain_topology *topo)
...
-			needs_rerun |= txw_fire(w, &w->txid, depth);
-	}
-	if (needs_rerun)
-		goto again;
+	do {
+		/* Iterating a htable during deletes is safe, but might skip entries. */
+		needs_rerun = false;
+		for (w = txwatch_hash_first(&topo->txwatches, &i);
+	    	w;
+	    	w = txwatch_hash_next(&topo->txwatches, &i)) {
```
Comment (wythe) [inline lightningd/watch.c]: nit: Above two lines should be tabbed over.  They are part of the for loop.

**Comment by wythe** [comment]
Yes this is a nice improvement.

File: lightningd/watch.c (hunk 2 above)
Comment (cdecker) [reply lightningd/watch.c]: Looks like they mix tabs and spaces

Comment (renepickhardt) [reply lightningd/watch.c]: I can quickly fix this again but in my editor it looked pretty good

Comment (cdecker) [reply lightningd/watch.c]: I know, but it doesn't in other editors that display tabs as 2, 4 or 8 spaces. Please configure your editor to use tabs instead :-)

**Comment by cdecker** [comment]
While we don't have a style-guide of our own we tend to rely heavily on the linux kernel guide (literally the first piece of information I got from @rustyrussell when I joined the project btw :wink:): https://www.kernel.org/doc/html/v4.10/process/coding-style.html

**Pull request merged**